File: examples/observability-cascade/observability-cascade-ruby-node/support.rb
# frozen_string_literal: true

require "pathname"

begin
  if ENV["OP_SDK_RUBY_PATH"] && !ENV["OP_SDK_RUBY_PATH"].empty?
    prebuilt_bundle = File.join(ENV["OP_SDK_RUBY_PATH"], "vendor", "bundle")
    if Dir.exist?(prebuilt_bundle)
      ENV["BUNDLE_PATH"] ||= prebuilt_bundle
      ENV["BUNDLE_DISABLE_SHARED_GEMS"] ||= "true"
    end
  end
  require "bundler/setup"
rescue LoadError
  nil
end

module CascadeNodeRuby
  ROOT = File.expand_path(__dir__)
  GENERATED_ROOT = File.expand_path("gen", ROOT)
  GEN_ROOT = File.expand_path("gen/ruby", ROOT)

  def self.find_repo_root(start)
    current = Pathname.new(start).expand_path
    loop do
      return current.to_s if current.join("sdk", "ruby-holons", "lib").directory?

      parent = current.parent
      raise "could not locate repository root" if parent == current

      current = parent
    end
  end

  SDK_LIB = File.join(find_repo_root(ROOT), "sdk", "ruby-holons", "lib")

  class << self
    def ensure_load_paths
      [ROOT, GENERATED_ROOT, GEN_ROOT, SDK_LIB].each do |path|
        $LOAD_PATH.unshift(path) unless $LOAD_PATH.include?(path)
      end
    end
  end
end

CascadeNodeRuby.ensure_load_paths

File: examples/observability-cascade/observability-cascade-python-node/support.py
from __future__ import annotations

import sys
from pathlib import Path

ROOT = Path(__file__).resolve().parent
GEN_ROOT = ROOT / "gen" / "python"


def find_repo_root(start: Path) -> Path:
    for candidate in [start, *start.parents]:
        if (candidate / "sdk" / "python-holons").is_dir():
            return candidate
    raise RuntimeError("could not locate repository root")


SDK_ROOT = find_repo_root(ROOT) / "sdk" / "python-holons"


def ensure_import_paths() -> None:
    for path in (ROOT, GEN_ROOT, SDK_ROOT):
        resolved = str(path)
        if resolved not in sys.path:
            sys.path.insert(0, resolved)
